openlr/openlr_decoder.cpp

```diff
@@ -191,4 +310,107 @@ void OpenLRDecoder::Decode(vector<LinearSegment> const & segments, uint32_t cons
   LOG(LINFO, ("Ambiguous routes:", allStats.m_moreThanOneCandidate));
   LOG(LINFO, ("Path is not reconstructed:", allStats.m_zeroCanditates));
 }
+
+void OpenLRDecoder::DecodeV2(vector<LinearSegment> const & segments,
+                             uint32_t const /* numThreads */, vector<DecodedPath> & paths)
+{
+  Graph graph(m_indexes.back(), make_unique<CarModelFactory>(m_countryParentNameGetterFn));
+
+  v2::Stats stat;
+  my::Timer timer;
+  RoadInfoGetter infoGetter(m_indexes.back());
+  for (size_t i = 0; i < segments.size(); ++i)
+  {
+    if (!DecodeOne(segments[i], m_indexes.back(), graph, infoGetter, paths[i], stat))
+      ++stat.m_routesFailed;
+    ++stat.m_routesHandled;
+
+    if (i % 100 == 0 || i == segments.size() - 1)
+    {
+      LOG(LINFO, (i, "segments are processed in",
+                  timer.ElapsedSeconds(),
+                  "seconds"));
+      timer.Reset();
+    }
+  }
+
+  LOG(LINFO, ("Total routes handled:", stat.m_routesHandled));
+  LOG(LINFO, ("Failed:", stat.m_routesFailed));
+  LOG(LINFO, ("No candidate lines:", stat.m_noCandidateFound));
+  LOG(LINFO, ("Wrong dnp:", stat.m_dnpIsZero));
+  LOG(LINFO, ("Wrong offsets:", stat.m_wrongOffsets));
+  LOG(LINFO, ("No shortest path:", stat.m_noShortestPathFound));
+}
+
+bool OpenLRDecoder::DecodeOne(LinearSegment const & segment, Index const & index, Graph & graph,
+                              RoadInfoGetter & infoGetter, DecodedPath & path, v2::Stats & stat)
+{
+  // TODO(mgsergio): Scale indexes.
+
+  graph.ResetFakes();
+
+  double const kPathLengthTolerance = 0.30;
```

Comment:
Please, move these consts to the top of the method body.

Reply:
Done.